src/processor/processor_task.cpp

```diff
@@ -18,6 +18,17 @@ void ProcessorTask::finalizeIfNecessary() {
     sink->finalize(executionContext);
 }
 
+static void addStructFieldsVectors(common::ValueVector* structVector, common::DataChunk* dataChunk,
+    storage::MemoryManager* memoryManager) {
+    auto structTypeInfo =
+        reinterpret_cast<common::StructTypeInfo*>(structVector->dataType.getExtraTypeInfo());
+    for (auto& childType : structTypeInfo->getChildrenTypes()) {
+        auto childVector = std::make_shared<common::ValueVector>(*childType, memoryManager);
+        dataChunk->addValueVector(childVector);
```

Comment:
Let's discuss the design choice here. I want to understand why we should keep childVector inside dataChunk.